neptune/new/internal/utils/__init__.py

```diff
@@ -170,3 +174,13 @@ def is_ipython() -> bool:
         return ipython is not None
     except ImportError:
         return False
+
+
+def expect_not_an_experiment(run: 'Run'):
+    from neptune.new.exceptions import NeptuneLegacyIncompatibilityException
+    try:
+        from neptune.experiments import Experiment
+        if isinstance(run, Experiment):
+            raise NeptuneLegacyIncompatibilityException()
+    except ImportError:
```

Comment:
Why do expect `ImportError`? It's part of the same repository.

Reply:
I'm trying to think about it as a legacy API, and protect against dropping it someday but leaving in this import? Dunno, really...

Reply:
Nah, :+1: to @shnela . In case of dropping the compatibility, we should be still aware of every "legacy" part in the codebase. It will be omitted someday instead of just trash it.

Reply:
ok, removed try-except

I also had to do local import due to circular dep (see https://github.com/neptune-ai/neptune-client/pull/611#discussion_r664337859). Maybe you'd prefer me to move this function to a new module (eg. `utils/compatibility.py`) ?

Reply:
Sure

Reply:
fixed, could you also reapprove https://github.com/neptune-ai/neptune-tensorflow-keras/pull/5 then? :)